AzureIotPnpDps.X/main.c

```diff
@@ -7,582 +7,1327 @@
-      az_result_failed(rc = az_json_writer_append_begin_object(&json_writer)) ||
-      az_result_failed(rc = az_json_writer_append_property_name(&json_writer, telemetry_name)) ||
-      az_result_failed(rc = az_json_writer_append_int32(
-                           &json_writer, current_device_temp)) ||
-      az_result_failed(rc = az_json_writer_append_end_object(&json_writer)))
-  {
-    return rc;
-  }
-  *out_payload = az_json_writer_get_bytes_used_in_destination(&json_writer);
-
-  return AZ_OK;
+	char reboot_delay[32];
+	az_json_writer jw;
+	az_json_reader jr;
+
+	debug_printInfo("  MAIN: %s() : Payload %s", __func__, az_span_ptr(payload));
+
+	if (az_span_size(payload) == 0 || (az_span_size(payload) == 2 && az_span_is_content_equal(empty_payload_span, payload)))
+	{
+		RETURN_ERR_IF_FAILED(build_error_response_payload(response, span_error_no_payload_reboot, out_response));
+		return AZ_ERROR_ITEM_NOT_FOUND;
+	}
+
+	RETURN_ERR_IF_FAILED(az_json_reader_init(&jr, payload, NULL));
+
+	while (jr.token.kind != AZ_JSON_TOKEN_END_OBJECT)
+	{
+		if (jr.token.kind == AZ_JSON_TOKEN_PROPERTY_NAME)
+		{
+			if (az_json_token_is_text_equal(&jr.token, span_command_reboot_delay))
+			{
+				RETURN_ERR_IF_FAILED(az_json_reader_next_token(&jr));
+				RETURN_ERR_IF_FAILED(az_json_token_get_string(&jr.token, reboot_delay, sizeof(reboot_delay), NULL));
+				break;
+			}
+		}
+		else if (jr.token.kind == AZ_JSON_TOKEN_STRING)
+		{
+			RETURN_ERR_IF_FAILED(az_json_token_get_string(&jr.token, reboot_delay, sizeof(reboot_delay), NULL));
+			break;
+		}
+		RETURN_ERR_IF_FAILED(az_json_reader_next_token(&jr));
+	}
+
+	if (reboot_delay[0] != 'P' || reboot_delay[1] != 'T' || reboot_delay[strlen(reboot_delay) - 1] != 'S')
+	{
+		debug_printError("  MAIN: Reboot Delay wrong format");
+		RETURN_ERR_IF_FAILED(build_error_response_payload(response, span_error_no_payload_reboot, out_response));
+		return AZ_ERROR_ARG;
+	}
+
+	reboot_delay_seconds = atoi((const char *)&reboot_delay[2]);
+
+	start_json_object(&jw, response);
+	append_jason_property_string(&jw, span_command_reboot_status, span_command_reboot_success);
+	append_jason_property_int32(&jw, span_command_reboot_delay, reboot_delay_seconds);
+	end_json_object(&jw);
+
+	*out_response = az_json_writer_get_bytes_used_in_destination(&jw);
+
+	return AZ_OK;
 }
 
-static int send_telemetry_message(void)
+/**********************************************
+* Handle commands
+**********************************************/
+static void handle_command_message(
+				az_span payload,
+				az_iot_pnp_client_command_request* command_request)
 {
-  int rc = az_iot_pnp_client_telemetry_get_publish_topic(
-              &pnp_client, AZ_SPAN_EMPTY, NULL, telemetry_topic, sizeof(telemetry_topic), NULL);
-  RETURN_ERR_WITH_MESSAGE_IF_FAILED(rc, "Could not get the telemetry topic");
+	az_span command_response_span = AZ_SPAN_FROM_BUFFER(commands_response_payload);
+	uint16_t return_code;
+	az_result azRet;
```

Comment:
nit: throughout the code base, the coding style is snake case
```suggestion
	az_result az_ret;
```

Reply:
changed all az_result variables to az_ret

[commit](https://github.com/daisukeiot/Microchip-PIC-IoT-Wx-1/commit/64508ae802742fd040c4263817d02b3a719ddeaa)